overlord/devicestate/devicestate.go

```diff
@@ -598,20 +583,12 @@ func Remodel(st *state.State, new *asserts.Model) (*state.Change, error) {
 		return nil, fmt.Errorf("cannot remodel to different series yet")
 	}
 
-	// TODO:UC20: support remodel, also ensure we never remodel to a lower
+	// TODO:UC20: ensure we never remodel to a lower
 	// grade
-	if !allowUC20RemodelTesting {
-		if current.Grade() != asserts.ModelGradeUnset {
-			return nil, fmt.Errorf("cannot remodel Ubuntu Core 20 models yet")
-		}
-		if new.Grade() != asserts.ModelGradeUnset {
-			return nil, fmt.Errorf("cannot remodel to Ubuntu Core 20 models yet")
-		}
-	} else {
-		// also disallows remodel from non-UC20 (grade unset) to UC20
-		if current.Grade() != new.Grade() {
-			return nil, fmt.Errorf("cannot remodel from grade %v to grade %v", current.Grade(), new.Grade())
-		}
+
+	// also disallows remodel from non-UC20 (grade unset) to UC20
+	if current.Grade() != new.Grade() {
+		return nil, fmt.Errorf("cannot remodel from grade %v to grade %v", current.Grade(), new.Grade())
```

Comment:
we maybe want a spefic error message for the pre-UC20 to UC20 case?

Reply:
This is done now.